src/stream.c

```diff
@@ -479,7 +479,9 @@ avifResult avifRWStreamWriteZeros(avifRWStream * stream, size_t byteCount)
 
 avifResult avifRWStreamWriteBits(avifRWStream * stream, uint32_t v, size_t bitCount)
 {
-    assert(((uint64_t)v >> bitCount) == 0); // (uint32_t >> 32 is undefined behavior)
+    if (bitCount < 32 && (v >> bitCount) != 0) {
+        return AVIF_RESULT_INVALID_ARGUMENT;
+    }
```

Comment:
If `bitCount >= 32`, then all `uint32_t` values can be represented, so no check is needed.

Should we disallow `bitCount == 0`?

Reply:
The standard probably allows it no? Let's keep it for now.

Reply:
Optional: Use `AVIF_ASSERT_OR_RETURN`. To me `AVIF_RESULT_INVALID_ARGUMENT` carries the meaning of an error from the user, not an implementation bug.

> Should we disallow `bitCount == 0`?

I do not mind enforcing it or not.

Reply:
Thanks for the suggestion. I found that `AVIF_ASSERT_OR_RETURN` makes the new test call `abort()` in the Debug build, so we can't use `AVIF_ASSERT_OR_RETURN` with a test for error checking. I also found that `avifRWStreamWriteVarInt()` returns `AVIF_RESULT_INVALID_PARAMETER` for a similar error. So in the end I just converted the if statement to use `AVIF_CHECKERR`.